.github/workflows/test.yml

```
@@ -10,6 +10,10 @@ on:
     # https://crontab.guru/#0_0_1_*_*
     - cron: "0 0 1 * *"
 
+concurrency:
+  group: ${{ github.workflow}}-${{ github.ref }}
```

Not sure if it is relevant, but should it be

```suggestion
  group: ${{ github.workflow }}-${{ github.ref }}
```

I have troubles to find out how the group expression is working from the official documentation (the only reference I found was [this](https://github.community/t/concurrency-cancel-in-progress-at-top-level-doesnt-seem-to-cancel-running-workflows/175801/5)), but wouldn't it be better to cancel all jobs on a brand with a bare `${{ github.ref }}`?

---

I guess if only some jobs are run, it makes a difference. With path filtering, that could be especially useful. Like if I push only a documentation change later on, it could only trigger a doc check, but not the main runs. If the old runs get canceled, the tests never run.

---

On the other hand, when the CI server is busy, the new jobs may not all start straight away, in which case it's useful if the first new job to start cancels all of the old ones. Maybe the best thing would be to remove the `{{ github.workflow }}` from test builds (i.e. all the workflows we have now) and try to remember to include it if we add a workflow for building documentation.
```suggestion
  group: ${{ github.ref }}
```